Content.Server/Chemistry/EntitySystems/TransformableContainerSystem.cs

```diff
@@ -0,0 +1,84 @@
+using Content.Server.Chemistry.Components;
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chemistry.EntitySystems;
+
+public sealed class TransformableContainerSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionsSystem = default!;
+    [Dependency] private readonly MetaDataSystem _metadataSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TransformableContainerComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<TransformableContainerComponent, SolutionChangedEvent>(OnSolutionChange);
+    }
+
+    private void OnMapInit(EntityUid uid, TransformableContainerComponent component, MapInitEvent args)
+    {
+        var meta = MetaData(uid);
+        if (string.IsNullOrEmpty(component.InitialName))
+        {
+            component.InitialName = meta.EntityName;
+        }
+        if (string.IsNullOrEmpty(component.InitialDescription))
+        {
+            component.InitialDescription = meta.EntityDescription;
```

Comment:
use _metadataSystem methods

Reply:
I'm not sure what you mean, this just stores the original name/description in the component so it can be reset later.

Reply:
my bad, got confused